pkg/operator/starter.go

```diff
@@ -654,3 +657,14 @@ func withAWSRegion(infraLister v1.InfrastructureLister) dc.DeploymentHookFunc {
 		return nil
 	}
 }
+
+func withHypershiftPriorityClass(isHypershift bool) dc.DeploymentHookFunc {
+	if !isHypershift {
+		return nil
+	}
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deployment.Spec.Template.Spec.PriorityClassName = hypershiftPriorityClass
+		return nil
+	}
+
+}
```

Comment:
IMO the `withHypershiftDeploymentHook` function would be a better place for this, what do you think?

Reply:
Agreed.

Initial, I saw a separate `withHypershiftReplicasHook`, so it looks like the idea was to have fine-grained hooks. But it looks weird to have them *that* fine grained.

Reply:
Fixed & pushed